### PR TITLE
fix: remove stream base class

### PR DESCRIPTION
**Pull request by marcosschroh**

### CODE REVIEW

File: kstreams/clients.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 logger = logging.getLogger(__name__)
 
-ConsumerType = TypeVar("ConsumerType", bound="Consumer")
-ProducerType = TypeVar("ProducerType", bound="Producer")
+KafkaConsumer = TypeVar("KafkaConsumer", bound="Consumer")
+KafkaProducer = TypeVar("KafkaProducer", bound="Producer")
```
Comment (Woile) [inline kstreams/clients.py:12]: Shouldn't we use the `Consumer` directly instead of this generic? What's the benefit?

Comment (Woile) [reply kstreams/clients.py:12]: On another note, `Consumer` is inheriting directly from `aiokafka.Consumer`, which makes it a "Kafka Consumer". Should the generic type be called different instead? `ConsumerT` and `ProducerT` for example?

Comment (marcosschroh) [reply kstreams/clients.py:12]: The change is just a rename. The benefit is that you could use a subclass of `Consumer` and `Producer` and `mypy` won't complaint. The `bound` is saying that `Stream.consumer` can be a `client.Consumer` or a subclass of it. The original idea was that end users can define their own consumers, this is the reason for it.

Comment (Woile) [reply kstreams/clients.py:12]: I get that, what I mean is that now the generic is "Kafka" specific (KafkaConsumer), while the class itself (Consumer) is not generic and is bound to kafka.

**Review by Woile — APPROVED** [review]

**Pull request merged**